site-src/concepts/choosing-route-types.md

```diff
@@ -0,0 +1,248 @@
+
+Example:
+
+```yaml
+kind: HTTPRoute
+spec:
+  parentRefs:
+  - name: gateway
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /api
+    backendRefs:
+    - name: api-backend
+      port: 8080
+```
+### GRPCRoute (Standard)
+
+Use this when your application uses gRPC and requires routing at the gRPC service or method level.
+
+This Route type requires HTTP/2 support on the Gateway.
+
+Example:
+
+```yaml
+kind: GRPCRoute
+spec:
+  parentRefs:
+  - name: gateway
+  rules:
+  - matches:
+    - method:
+        service: payments.Service
+        method: Charge
+    backendRefs:
+    - name: payments-backend
+      port: 50051
+```
+### TLSRoute (Promoted)
+
+Use this when your application uses TLS and you want the Gateway to pass encrypted traffic directly to the backend without decrypting it.
+
+Common scenarios include:
+
+- End-to-end TLS encryption
+- Applications that manage their own certificates
+- Routing based only on the SNI (TLS hostname)
+
+TLSRoute does not support HTTP-level routing such as paths or headers.  
+Use this only when the Gateway should not terminate TLS.
+
+Example:
+
+```yaml
+kind: TLSRoute
+spec:
+  parentRefs:
+  - name: gateway
+  rules:
+  - matches:
+    - snis: ["secure.example.com"]
+    backendRefs:
+    - name: secure-app
+      port: 8443
+```
+### TCPRoute (Experimental)
+
+Use this when your application communicates using raw TCP.
+
+Common examples include:
+
+- Databases such as Postgres or MySQL
+- Message brokers such as MQTT or AMQP
+- Mail protocols like SMTP or IMAP
+
+TCPRoute provides simple Layer-4 forwarding without HTTP or TLS inspection.
+
+Example:
+
+```yaml
+kind: TCPRoute
+spec:
+  parentRefs:
+  - name: gateway
+  backendRefs:
+  - name: postgres
+    port: 5432
+```
+### UDPRoute (Experimental)
+
+Use this when your application uses UDP.
+
+Common examples include:
+
+- DNS servers
+- Game servers
+- RTP or other streaming workloads
+
+UDPRoute forwards packets at Layer 4 without connection state or protocol inspection.
+
+Example:
+
+```yaml
+kind: UDPRoute
+spec:
+  parentRefs:
+  - name: gateway
+  backendRefs:
+  - name: dns-backend
+    port: 53
+```
+## 3. Choosing the Right Route — Decision Table
+
+The table below provides a quick reference for selecting the appropriate Route type based on your application's protocol and routing needs.
+
+| Application Type       | Route Type | Reason                        |
+|------------------------|------------|-------------------------------|
+| Website / REST API     | HTTPRoute  | Layer-7 routing, TLS termination |
+| gRPC service           | GRPCRoute  | Service and method-based routing |
+| HTTPS passthrough      | TLSRoute   | Backend terminates TLS          |
+| Database               | TCPRoute   | Raw TCP protocol                 |
+| DNS                    | UDPRoute   | Uses UDP                        |
+| Game server            | UDPRoute   | Uses UDP                        |
+
```

Comment:
You could also reuse the Route summary table from https://gateway-api.sigs.k8s.io/concepts/api-overview/#route-summary-table, which includes the "Routing Discriminator" column - the idea of Routing Discriminators is really important for understanding which Route to use.